### PR TITLE
audit log v1

### PR DESCRIPTION
**Pull request by bendecoste**
Closes capeprivacy/cape#<issue>

### CODE REVIEW

File: models/audit.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package models
+
+import (
+	"fmt"
+	"github.com/mitchellh/mapstructure"
+	"io"
+	"k8s.io/apimachinery/pkg/util/json"
```
Comment (justin1121) [inline models/audit.go:7]: I don't think this is the right json package...

File: coordinator/audit/pg_logger.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+)
+
+var _ Logger = &PgLogger{}
+
+type PgLogger struct {
+	DB db.AuditLogDB
+}
+
+func (p *PgLogger) Log(name models.AuditEventName, data ...interface{}) error {
+	if len(data)%2 != 0 {
```
Comment (justin1121) [inline coordinator/audit/pg_logger.go:17]: Why not just make label an actual argument and then allow n extra data?

File: models/audit.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package models
```
Comment (kitschysynq) [inline models/audit.go:1]: This should go into the `audit` package.

File: coordinator/audit/pg_logger.go (hunk 2 above)
Comment (bendecoste) [reply coordinator/audit/pg_logger.go:17]: You can have multiple labels, e.g. `"label1", object, "label2", object, "label3", object`

File: models/audit.go (hunk 3 above)
Comment (bendecoste) [reply models/audit.go:1]: iirc that caused a circular dependency, will check again

**Review by kitschysynq — APPROVED** [review]

**Pull request merged**